src/core/featuremodel.cpp

```diff
@@ -850,31 +849,50 @@ class MatchCollectingFilter : public QgsPointLocator::MatchFilter
     }
 };
 
-void FeatureModel::applyVertexModelToLayerTopography()
+void FeatureModel::applyVertexModelTopography()
 {
   if ( !mVertexModel )
     return;
 
-  QgsPointLocator *loc = new QgsPointLocator( mLayer );
   const QVector<QPair<QgsPoint, QgsPoint>> pointsMoved = mVertexModel->verticesMoved();
-  for ( const auto &point : pointsMoved )
+  const QVector<QgsPoint> pointsDeleted = mVertexModel->verticesDeleted();
+
+  const QVector<QgsVectorLayer *> vectorLayers = QgsProject::instance()->layers<QgsVectorLayer *>();
```

Comment:
Can we pass the project in as property? It's already available in qml.

Reply:
@m-kuhn , sure, I can change that in a separate PR (after we merge the digitizing logger PR to avoid conflicts).